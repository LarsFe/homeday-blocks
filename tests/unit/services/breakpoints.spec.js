import { mount } from '@vue/test-utils';
import {
  setBreakpoints,
  getBreakpoints,
  Responsive,
} from '@/services/breakpoints';

const BREAKPOINTS = {
  s: '(max-width:599px)',
  m: '(min-width:600px)',
  l: '(min-width:900px)',
  xl: '(min-width:1200px)',
};

describe('Breakpoints service', () => {
  let wrapper;

  afterEach(() => {
    // We reset the breakpoints after each test
    setBreakpoints({});
  });

  it('can set the breakpoints', () => {
    setBreakpoints(BREAKPOINTS);
    expect(getBreakpoints()).toBe(BREAKPOINTS);
  });

  describe('Responsive component', () => {
    it('renders no markup', () => {
      wrapper = mount(Responsive);
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('provides "matches" object as part of a scoped slot', () => {
      setBreakpoints(BREAKPOINTS);
      wrapper = mount(Responsive, {
        scopedSlots: {
          default: '<div>{{Object.keys(props.matches)}}</div>',
        },
      });
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('provides "indeterminate" flag as part of a scoped slot', () => {
      setBreakpoints(BREAKPOINTS);
      wrapper = mount(Responsive, {
        scopedSlots: {
          default: '<div>{{props.indeterminate}}</div>',
        },
      });
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
