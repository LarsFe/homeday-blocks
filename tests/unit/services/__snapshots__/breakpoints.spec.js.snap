// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breakpoints service Responsive component provides "indeterminate" flag as part of a scoped slot 1`] = `<div>true</div>`;

exports[`Breakpoints service Responsive component provides "matches" object as part of a scoped slot 1`] = `
<div>[
  "s",
  "m",
  "l",
  "xl"
  ]</div>
`;

exports[`Breakpoints service Responsive component renders no markup 1`] = `undefined`;
